File: src/backend/cs/StudyRoomSystem/StudyRoomSystem.Core/Structs/Seat.cs
namespace StudyRoomSystem.Core.Structs;

public class Seat
{
    public required Guid Id { get; set; }
    public required Guid RoomId { get; set; }
    public required int Row { get; set; }
    public required int Col { get; set; }

    public virtual Room Room { get; set; } = null!;
}

File: src/backend/Java/StudyRoomSystem/src/main/java/com/zyx/studyroomsystem/pojo/Seat.java
package com.zyx.studyroomsystem.pojo;

import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

import java.util.UUID;

@Data
public class Seat {

    private UUID id;

    @NotNull(message = "房间ID不能为空")
    private UUID roomId;

    @NotNull(message = "行号不能为空")
    @Min(value = 1, message = "行号必须大于等于 1")
    private Integer row;

    @NotNull(message = "列号不能为空")
    @Min(value = 1, message = "列号必须大于等于 1")
    private Integer col;
}
